#!usr/bin/env python3
PACKAGE = "vg_gpn"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()
# name, type, level, help, default, min, max
gen.add("threshold_red",   double_t, 0, "Threshold for the red channel",   0.0, 0.0, 255.0)
gen.add("threshold_green", double_t, 0, "Threshold for the green channel", 0.0, 0.0, 255.0)
gen.add("threshold_blue",  double_t, 0, "Threshold for the blue channel",  0.0, 0.0, 255.0)

gen.add("min_threshold_red",  double_t, 0, "min Threshold for the red channel",  0.0, 0.0, 255.0)
gen.add("min_threshold_green",  double_t, 0, "min Threshold for the green channel",  0.0, 0.0, 255.0)
gen.add("min_threshold_blue",  double_t, 0, "min Threshold for the blue channel",  0.0, 0.0, 255.0)

gen.add("max_threshold_red",  double_t, 0, "max Threshold for the red channel",  0.0, 0.0, 255.0)
gen.add("max_threshold_green",  double_t, 0, "max Threshold for the green channel",  0.0, 0.0, 255.0)
gen.add("max_threshold_blue",  double_t, 0, "max Threshold for the blue channel",  0.0, 0.0, 255.0)



exit(gen.generate(PACKAGE, "vg_gpn", "Threshold"))